include/deal.II/base/point.h

```diff
@@ -560,6 +560,50 @@ inline DEAL_II_CUDA_HOST_DEV
 
 
 
+namespace internal
+{
+  namespace PointImplementation
+  {
+    template <int dim,
+              typename Number,
+              typename OtherNumber,
+              typename std::enable_if<
+                !std::is_integral<
+                  typename ProductType<Number, OtherNumber>::type>::value,
+                int>::type = 0>
+    inline DEAL_II_CUDA_HOST_DEV
+      Point<dim, typename ProductType<Number, OtherNumber>::type>
+      division_operator(const Point<dim, Number> &p, const OtherNumber factor)
+    {
+      using common_type = typename ProductType<Number, OtherNumber>::type;
+      Point<dim, common_type> tmp;
+      const auto              inverse_factor =
+        static_cast<common_type>(1.) / static_cast<common_type>(factor);
+      for (unsigned int i = 0; i < dim; ++i)
+        tmp[i] = p[i] * inverse_factor;
+      return tmp;
+    }
+
+    template <int dim,
+              typename Number,
+              typename OtherNumber,
+              typename std::enable_if<
+                std::is_integral<
+                  typename ProductType<Number, OtherNumber>::type>::value,
+                int>::type = 0>
+    inline DEAL_II_CUDA_HOST_DEV
+      Point<dim, typename ProductType<Number, OtherNumber>::type>
+      division_operator(const Point<dim, Number> &p, const OtherNumber factor)
+    {
+      Point<dim, typename ProductType<Number, OtherNumber>::type> tmp;
+      for (unsigned int i = 0; i < dim; ++i)
+        tmp[i] = p[i] / factor;
```

Comment:
What's the reason to not pull out the inverse of the factor here? Is it more efficient to divide by an integer than to multiply by a floating point number?

Reply:
I had been wondering the same thing wrt. all of the refactoring that's been done here, but I thought that I must be missing something. Couldn't we forward all of the operations for `Point::operator/=` and `Tensor::operator/=` to their respective `operator*=`?

Reply:
This is the case where both the `ProductType` for `Number` and `OtherNumber` is integral. Hence, `1/factor` is zero and it seems that we can't use the same implementation.